### ISSUE TITLE
LNURLw support

### ISSUE THREAD

**Issue by leesalminen** [lawalletio/lawallet-web#4]
Hi there,

We've been testing out the wallet for Bolt Card support, looks great!

We couldn't find a way to redeem LNURLw vouchers with the wallet though. Can you confirm if there is a way currently? If not, is is something you are willing to support? I may have some time to PR you if needed.

**Comment by cuervoshi** [comment]
Hi Lee!
LNURLw support is currently available. If you scan a LNURLw with the wallet web app, it should take you directly to the claim process. Alternatively, you can paste the LNURLw in the transfer section, and it should guide you to the claim as well.

If this isn't working as expected, please let us know
Looking forward to your feedback.

**Comment by leesalminen** [comment]
Hmm, we did try it a couple ways and didn't seem to be working as expected. I took a look in the code [0] and saw that it never checks for `TransferTypes.LNURL`.

[0] https://github.com/lawalletio/lawallet-web/blob/37ede0dcf72759203e8f3ff360eb1074093ddad6/src/app/%5Blng%5D/scan/page.tsx#L96

**Comment by leesalminen** [comment]
Hmm, we did try it a couple ways and didn't seem to be working as expected. Will keep trying and report back.

**Comment by leesalminen** [comment]
Yeah, we just tried with 3 LNURLw QR codes from 3 different providers, didn't work. Could you give it a try on your end? Maybe we're just being dumb?

**Comment by cuervoshi** [comment]
> Yeah, we just tried with 3 LNURLw QR codes from 3 different providers, didn't work. Could you give it a try on your end? Maybe we're just being dumb?

Could you tell me which provider you are using? I have only tested with the LNURLw extension from LNBits. Maybe they use different formats.

**Comment by leesalminen** [comment]
We've tried with Blink, Bull Bitcoin & Azteco

**Comment by leesalminen** [comment]
Let me try fresh, maybe they expired or something.

**Comment by leesalminen** [comment]
Here's a LNURLw for 1 EUR, can't scan it from my lawallet-web.vercel.app

![Uploading Screenshot 2024-12-19 at 1.07.38 PM.png…]()

**Comment by leesalminen** [comment]
![Screenshot 2024-12-19 at 1 06 13 PM](https://github.com/user-attachments/assets/71fb0d21-c071-4a8d-b006-cfa55c26cb33)

**Comment by cuervoshi** [comment]
Now I understand the problem.
The request to the decoded LNURL 
`https://lnurlv3.bullbitcoin.com/withdrawRequest?s=...`
is resulting in a CORS request error.

I will try to refactor the way the request is sent to solve this problem.
Thank you very much for the report! I will keep you informed through this issue.

**Comment by leesalminen** [comment]
Aha, CORS! OK, cool, I will also report to Bull Bitcoin as well. But would be good if your wallet could make the request from the backend as probably others also have this issue.

**Comment by leesalminen** [comment]
Confirmed it was CORS. I got Bull Bitcoin to set up CORS on the LNURLw endpoints and now works with La Wallet.

**Comment by cuervoshi** [comment]
> Confirmed it was CORS. I got Bull Bitcoin to set up CORS on the LNURLw endpoints and now works with La Wallet.

Good news!
I've also implemented a proxy on requests.
This feature is currently available in the add-proxy branch if you want to try it out. I'll continue to refine it before releasing it to production to make sure everything works smoothly.
Thanks for your report
